lib/matplotlib/tests/test_contour.py

```diff
@@ -837,3 +837,11 @@ def test_allsegs_allkinds():
         assert len(result) == 2
         assert len(result[0]) == 5
         assert len(result[1]) == 4
+
+
+def test_contourf_rasterize():
+    fig, ax = plt.subplots()
+    data = [[0, 1], [1, 0]]
+    circle = mpatches.Circle([0.5, 0.5], 0.5, transform=ax.transAxes)
+    cs = ax.contourf(data, clip_path=circle, rasterized=True)
+    assert cs._rasterized
```

Comment:
is checking `_rasterized` enough, or should I check against a baseline image?

Reply:
I would have a baseline pdf for this.  You could make the raster coarse (25 dpi) to make the pixels obvious.  I'm actually curious if this change actually does something, so Id recommend also testing it.